src/router.ts

```diff
@@ -75,9 +73,29 @@ export class Router {
         return;
       }
 
+      const currentPrice = getPythPrice(new PublicKey(dip_deposit.splToken));
+      const fractionOfYear = (Date.now() - dip_deposit.expirationMs) / 365 * 24 * 60 * 60 * 1_000;
+      const vol = THEO_VOL_MAP[dip_deposit.splToken] * 1.2;
+      const bvePrice = blackScholes(currentPrice, dip_deposit.strike / 1_000_000, fractionOfYear, vol, 0.01, 'call');
+
+      const price = order["price"];
+
+      if (bvePrice > price) {
+        // If the price is worse than the BVE, then do not use the MM, treat it
+        // like there is no MM bid.
+        console.log("Not routing to MM due to price:", bvePrice, price);
```

Comment:
does this only attempt to route to a MM upon rerun or any time a new price comes from a MM?

Reply:
Only on rerun. We assume that every 10 minutes or every deposit, whichever is sooner, will catch the new price from MM.
I dont think we want to update for every time an MM updates since I dont want to encourage them to trade at higher frequencies and DOS us.

Reply:
In the case that they have a bid live though we will immediately route the DIP risk to them though correct?

Reply:
Correct.